Adding variables for SCM info

samdmarshall opened the pull request:
**Title**: Adding ability to query the SCM for info about the environment

---

Fixes: #50 

**Description**:
- [x] Add new argument flag with some default SCM options for detecting information
  - d8e964052ab29837a57a4b92d7e01c20bd2b56a7
- [x] Add new module for fetching the SCM info
  - ed9878076cc8d774658cbeab2df11242161fe156
- [x] Define how the SCM info will be serialized
  - ed9878076cc8d774658cbeab2df11242161fe156
- [x] Add support for Mercurial
  - b5cd6c943dc9e5ae1f97cdbec6d3a7cdeb927b88
- [x] Correctly handle serialization of the `scm-versions.xcconfig`
  - 96ac39363275ca24d61299b5159cbdd1d92368ff
- [x] Add documentation for new flag and behavior
  - 731f4fa59ddd178a3170064a2d5bb4594f8d996e
- [x] Add unit tests to cover generating the `scm-versions.xcconfig` for all supported SCMs
  - e72318d445b924bf6d06773ef31de6a1eead7f62

samdmarshall committed:
adding new flag

samdmarshall-danger commented:
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Changes to build files</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Changes to installation requirements files</td>
    </tr>
  </tbody>
</table>

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true"><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/247/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/247/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></td>
    </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/246/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/246/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/245/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/245/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/243/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/243/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/242/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/242/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/241/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/241/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/240/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/240/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
      <tr>
        <td>:white_check_mark:</td>
        <td data-sticky="true"><del><a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/239/artifacts/0/$CIRCLE_ARTIFACTS/htmlcov/index.html">html coverage report</a> &amp; <a href="https://circleci.com/api/v1/project/samdmarshall/pyconfig/239/artifacts/0/$CIRCLE_ARTIFACTS/lint_output.txt">linter report</a></del></td>
      </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

samdmarshall committed:
adding git and svn support

samdmarshall committed:
adding mercurial support

Note: currently doesn't export the build because of attempting to serialize the scm-versions.xcconfig

samdmarshall committed:
updating code due to linting errors

samdmarshall committed:
skipping analysis for auto-generated file

samdmarshall committed:
adding scm info unit tests

samdmarshall committed:
update to requirement on mercurial and subversion tools

samdmarshall committed:
updating readme and docs

samdmarshall committed:
fixing bug in analysis engine
